pkg/util/file_indexer.go

```diff
@@ -83,6 +85,75 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the contex
+func gitIgnoreFilePath(directory string) (string, error) {
+	directoryFile, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+
+	switch mode := directoryFile.Mode(); {
+	case mode.IsDir():
+		// whether the mode is a directory stuff
+		// mode directory bit is set
+		return filepath.Join(directory, ".gitignore"), nil
+	case mode.IsRegular():
+		// whether the mode is a file stuff
+		// No mode type is set
+		return filepath.Join(filepath.Dir(directory), ".gitignore"), nil
+	}
+
+	return directory, nil
+}
+
+// Check .gitignore file exists or not
+// Add odo-file-index.json entry to .gitignore
+func checkGitIgnoreFile(directory string) error {
+
+	ignoreFile, err := gitIgnoreFilePath(directory)
+	_, err = os.Stat(ignoreFile)
+
+	// .odo/odo-file-index.json file should be created in .odo
+	// open, read and write .odo/odo-file-index.json to .gitignore file if exists
+	// if .gitignore does not exist then create the file and write .odo/odo-file-index.json to it
+	if err == nil {
+		file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
+		if err != nil {
+			return errors.Wrap(err, "failed opening file")
+		}
+		defer file.Close()
+
+		data, err := ioutil.ReadFile(ignoreFile)
+		if err != nil {
+			return errors.Wrap(err, "failed reading data from file")
+		} else {
+			// check whether .odo/odo-file-index.json is already in the .gitignore file
+			if !strings.Contains(string(data), ".odo/odo-file-index.json") {
+				if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
+					return errors.Wrap(err, "failed writing string to file")
+				}
+			}
+		}
+	} else if os.IsNotExist(err) {
+		files, err := ioutil.ReadDir(filepath.Join(directory))
+		if err != nil {
+			return errors.Wrap(err, "No such directory")
+		}
+		if len(files) > 1 {
+			file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+			if err != nil {
+				return errors.Wrap(err, "failed opening file")
+			}
+			defer file.Close()
+
+			if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
+				return errors.Wrap(err, "failed writing string to file")
```

Comment:
Which string? Which file? We could rephrase this to something like `Failed to add odo-file-index.json to gitignore`. That's just a suggestion. You could, of course, change it to something else that conveys the user what *exactly* failed.

Reply:
For better understanding to the user I have done the modification in my latest change update.